### PR TITLE
SHRTCODE (Independent Publisher)

### PR DESCRIPTION
**Pull request by sekharmalla**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by sekharmalla** [comment]
Hi @v-EgorKozhadei, Thank you.

about the other connectors getting affected:- That connector has also been submitted for me and @harshinivarma, I have just updated the contact email address in that, and nothing changed in the API definition or settings file. 

Do let me know if it's possible to change or can't be done.

Thank you,
Chandra

**Comment by v-EgorKozhadei** [comment]
Hi @sekharmalla!

Thank you for the answer! I think it'd be better to update an existing connector in another PR.
In this PR we will review the new connector and queue it up for certification and deployment once everything is done here.

Could you please raise a PR with an update for the other connector?

Let us know if you have any questions, thank you!

**Comment by sekharmalla** [comment]
Hello @v-EgorKozhadei, thank you for your response, 

It's okay i will update that PR sometime later as i have few updates coming to that connector as well, thanks and looking forward for certifying the connector.

**Comment by sekharmalla** [comment]
Hello just checking on the connector status, do I have to make any changes? 

please do let me know if yes,

Regards,

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "contact": {
       "name":"Chandra Sekhar & Harshini Varma",
       "url":"https://github.com/sekharmalla",
-      "email":"Sekharmalla.vfx@gmail.com"
+      "email":"chandra.malla@msftcommunity.com"
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/Notion/apiDefinition.swagger.json:10]: @sekharmalla please move changes for other connectors to separate PRs.

File: independent-publisher-connectors/SHRTCODE/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                  "type": "object",
+                  "properties": {
+                    "code": {
+                      "type": "string",
+                      "description": "code",
+                      "title": "Code"
+                    },
+                    "short_link": {
+                      "type": "string",
+                      "description": "short_link",
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/SHRTCODE/apiDefinition.swagger.json:45]: @sekharmalla descriptions should look like valid sentences. You can refer to [the documentation](https://docs.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#parameters).

File: independent-publisher-connectors/SHRTCODE/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                  },
+                  "description": "result"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Shortening a Link",
+        "description": "Create a short link for a given URL. Requires a `url` parameter.",
+        "operationId": "ShorteningaLink",
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/SHRTCODE/apiDefinition.swagger.json:87]: @sekharmalla operationId property should be in pascal case. You can refer to [this example](https://docs.microsoft.com/en-us/connectors/custom-connectors/coding-standards#actions-and-triggers).

File: independent-publisher-connectors/SHRTCODE/settings.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/SHRTCODE/settings.json:1]: @sekharmalla please remove this file from PR, it's not required.

File: independent-publisher-connectors/SHRTCODE/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "properties": {
+    "connectionParameters": {},
+    "iconBrandColor": "#da3b01",
+    "capabilities": [],
+    "publisher": "Chandra Sekhar Malla"
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/SHRTCODE/apiProperties.json:6]: @sekharmalla stackOwner property is missing. You can find more information in [this template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json).

File: independent-publisher-connectors/SHRTCODE/readme.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# SHRTCODE
+the privacy-friendly URL Shortener. With this free Link Shortener connector you can make Links shorter and easier to remember.
+
+## Publisher
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/SHRTCODE/readme.md:4]: @sekharmalla this heading should look like this: `Publisher: Publisher's Name`. You can refer to [readme.md example](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md?plain=1). All of the required headings should be present.

**Review by v-EgorKozhadei — COMMENTED** [review]
Hi @sekharmalla!

Please check comments after the review. There are just a few minor things that need to be fixed.
The most important one is having anther connector's file in the PR. Please raise a separate PR for each independent publisher connector if you want to update them.

Please let us know if you have any questions.

**Review by sekharmalla — COMMENTED** [review]
Hi @v-EgorKozhadei 
I think i have fixed all the mentioned changes. do let me know should you require further changes, also can delete apiDefinition.swagger.json file so that it will be out of this PR?

kindly do let me know,

Thanks,

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json (hunk 1 above)
Comment (sekharmalla) [reply independent-publisher-connectors/Notion/apiDefinition.swagger.json]: @v-EgorKozhadei 
Please do let me know if I can delete this file or how should I move this to a new PR. 

Thanks

**Comment by v-EgorKozhadei** [comment]
Hi @sekharmalla!

Thank you very much for the update!
Please make sure that apiDefinition file for Notion connector is not affected in any way so that it is not included in changed files for this PR. You can raise an update for this connector in a separate PR.

**Comment by sekharmalla** [comment]
Hi @v-EgorKozhadei 

I have kept the file as it is and did not perform any changes, if it's possible can you check from your side and do let me know if I have to delete the file from this PR.

right now it's showing like this, 

image

**Comment by v-EgorKozhadei** [comment]
Hi @sekharmalla!

Thank you for your message!
We should remove the file from the PR, yes. You could try replacing the contents with the original ones that are in the open-source repository, the link to the file [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/independent-publisher-connectors/Notion/apiDefinition.swagger.json). This file should not show up in "Files changed" tab.

**Comment by sekharmalla** [comment]
Hi @v-EgorKozhadei 

Thank you for your message.

I have replaced content with original ones but it's still there in files changed. Please do let me know what should i do now and sorry for the trouble.

image


Thanks

**Comment by v-EgorKozhadei** [comment]
Hi @sekharmalla!

Could you please try to go to "Changed files" tab, click at three dots icon at the right upper corner of the file, click on "Edit file". I see that there's a newline at the end of the file. Try to remove it and click on "Preview changes" tab. Removing the newline should make the file match the original one.

**Comment by sekharmalla** [comment]
Hi @v-EgorKozhadei 

I have removed the space and checked in the preview but it's showing like this even after committing the file. 

image

Thanks,

**Comment by v-EgorKozhadei** [comment]
Hi @sekharmalla!

It should be displayed as unchanged. I tried comparing both files and all characters are the same now. There must be something wrong with the formatting, but files look the same for me.

Could you please quickly recreate your fork and raise a new PR with SHRTCODE connector files only? I'm sorry for the inconvenience.

**Comment by sekharmalla** [comment]
Hi @v-EgorKozhadei 

Sure, I will now open New PR with SHRTCODE and leave this PR.

Thanks,

**Comment by v-EgorKozhadei** [comment]
Hi @sekharmalla!

Thank you very much! I'll close this one when the new one will be opened.

**Comment by sekharmalla** [comment]
@v-EgorKozhadei 

I am not able to make new PR as it's showing 'is able to merge' 

image


is it possible you can refer me to any guide that I can follow?

Thanks

**Comment by v-EgorKozhadei** [comment]
@sekharmalla 
I'll try to push one change to your branch to see if it fixes the problem.

If it won't help, we can recreate the fork: delete the existing one by going to Settings tab in your fork, create a new fork from the open-source repository by clicking on "Fork", clone new fork, add and commit connector files, then create a new PR.

**Comment by v-EgorKozhadei** [comment]
@sekharmalla 
Thank you for your patience.
The issue is fixed, everything is OK now.

**Comment by sekharmalla** [comment]
Thank you very much you are awesome @v-EgorKozhadei 👍🏼

**Comment by v-EgorKozhadei** [comment]
@sekharmalla

Congratulations, your connector is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.  

Please let us know if you have any questions. 

Thank you very much for working with us.

**Comment by v-EgorKozhadei** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @sekharmalla,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @sekharmalla,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]
